dfsclient/src/main/java/org/hpccsystems/dfs/client/RowServiceOutputStream.java

```diff
@@ -254,7 +320,15 @@ private static class RowServiceResponse
         }
         catch (IOException e)
         {
-            throw new HpccFileException("Failed on initial remote read read trans", e);
+            HpccFileException wrappedException = new HpccFileException("Failed on initial remote read read trans", e);
+            if (writeSpan != null)
+            {
+                Attributes attributes = Attributes.of(ServerAttributes.SERVER_ADDRESS, this.rowServiceIP,
```

Comment:
I noticed this seems to be repeated in a few places, why not add a function in utils to handle the attributes annotation in a single place. Also, why aren't we annotating the span as soon as it's created?

Reply:
I want to cleanup how I am handling adding events / exceptions in the next PR, so I was holding off on cleaning up how attributes get added for those events. However, those attributes should have been added to the span when it was originally created. I will do that as part of this PR.